test/concurrent.cc

```diff
@@ -778,34 +842,34 @@ TEST(ConcurrentTest, DownstreamDoneBothEventualsSuccess) {
 
   std::vector<std::string> values = {"1", "2"};
 
-  EXPECT_THAT(values, Contains(future.get()));
+  this->ExpectOrContains(std::move(future), values);
```

Comment:
Let's just be explicit here and do the `constexpr` here. You can use `TypeParam` in the test to get access to the type parameter, thus:
```suggestion
  if constexpr (std::is_same_v<TypeParam, ConcurrentType>) {
    EXPECT_THAT(values, Contains(future.get()));
  } else {
    EXPECT_EQ(values[0], future.get());
  }
```